File: binary_search.py
from string import printable
from unicodedata import numeric


def binary_search(arr , number):
  length = len(arr)
  mid = length/2
  for i in range(0 , mid):
    currentPos = i
    oppositePos = length - i - 1
    if arr[currentPos] == number or arr[oppositePos] == number:
      return 1
  if arr[mid] == number:
    return 1
  return 0

print(binary_search([1,2,3,4,5], 5))

File: binary_search.js
const arr = [1,2,3,4,5]
function binarySearch(searchValue, l, r)
{
  if(!searchValue){
    return -1
  }
  const m = parseInt((l + r) / 2)
  if((m === l || m === r) && arr[m] !== searchValue) {
    return -1
  }
  if(searchValue < arr[m]){
     return binarySearch(searchValue , l , m)
  }
  if(searchValue > arr[m]){
     return binarySearch(searchValue, m , r)
  }
  return 1
}
console.log(binarySearch(3, 0,arr.length))
